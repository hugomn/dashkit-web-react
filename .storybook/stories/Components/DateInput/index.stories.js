import React from "react";
import { storiesOf } from "@storybook/react";
import DateInput from "../../../../src/components/DateInput/template";

storiesOf("Components/DateInput", module).add("Default input", () => (
  <DateInput name={"date"} disabled={false} value={new Date()} onSelect={() => {}} />
));
