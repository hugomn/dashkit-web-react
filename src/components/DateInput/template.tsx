import { Calendar, CalendarProps, DropButton, TextInput } from "grommet";
import moment from "moment";
import React, { useState } from "react";
import { handleSelect, toggleOpen } from ".";

interface IProps {
  disabled?: HTMLInputElement["disabled"];
  name?: HTMLInputElement["name"];
  onSelect?: CalendarProps["onSelect"];
  value: HTMLInputElement["value"];
}

const DateInput: React.FC<IProps> = ({ disabled, name, onSelect, value }) => {
  const [open, setOpen] = useState(false);
  const [date, setDate] = useState<string | undefined>(value);
  return (
    <DropButton
      open={open}
      onClose={toggleOpen(false, setOpen)}
      onOpen={toggleOpen(true, setOpen)}
      disabled={disabled}
      dropContent={<Calendar date={date} onSelect={handleSelect(setDate, setOpen, onSelect)} />}>
      <TextInput
        name={name}
        disabled={disabled}
        value={moment(date).format("DD/MM/YYYY")}
        data-testid="input-date"
      />
    </DropButton>
  );
};

export default DateInput;
