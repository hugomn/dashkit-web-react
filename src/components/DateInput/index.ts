import { Dispatch, SetStateAction } from "react";
import template from "./template";

export const toggleOpen = (open: boolean, setOpen: Dispatch<SetStateAction<boolean>>) => () => {
  setOpen(open);
};

export const handleSelect = (
  setDate: Dispatch<SetStateAction<string | undefined>>,
  setOpen: Dispatch<SetStateAction<boolean>>,
  onSelect?: any
) => (date: any) => {
  setDate(date);
  setOpen(false);
  if (onSelect) {
    onSelect(date);
  }
};

export default template;
