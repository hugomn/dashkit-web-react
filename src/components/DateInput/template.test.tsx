import { render } from "@testing-library/react";
import React from "react";
import DateInput from "./template";
import { handleSelect, toggleOpen } from ".";

describe("DateInput component", () => {
  const { container } = render(<DateInput value="2020-02-02" />);
  it("should match the snapshot", () => {
    expect(container).toMatchSnapshot();
  });

  it("handleSelect should call the correct handlers", () => {
    const mockSetDate = jest.fn();
    const mockSetOpen = jest.fn();
    const mockOnSelect = jest.fn();
    const mockDate = "2020-01-02";
    handleSelect(mockSetDate, mockSetOpen, undefined)(mockDate);
    expect(mockSetDate).toHaveBeenLastCalledWith(mockDate);
    expect(mockSetOpen).toHaveBeenLastCalledWith(false);
    expect(mockOnSelect).not.toBeCalled();
    handleSelect(mockSetDate, mockSetOpen, mockOnSelect)(mockDate);
    expect(mockOnSelect).toHaveBeenLastCalledWith(mockDate);
  });

  it("toggleOpen should call correctly the callback", () => {
    const mockSetOpen = jest.fn();
    toggleOpen(true, mockSetOpen)();
    expect(mockSetOpen).toHaveBeenLastCalledWith(true);
    toggleOpen(false, mockSetOpen)();
    expect(mockSetOpen).toHaveBeenLastCalledWith(false);
  });
});
