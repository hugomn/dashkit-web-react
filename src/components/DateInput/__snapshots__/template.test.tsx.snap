// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput component should match the snapshot 1`] = `
<div>
  <button
    aria-label="Open Drop"
    class="StyledButton-sc-323bzc-0 jfbKsv"
    type="button"
  >
    <div
      class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
    >
      <input
        autocomplete="off"
        class="StyledTextInput-sc-1x30a0s-0 kvYrcH"
        data-testid="input-date"
        value="02/02/2020"
      />
    </div>
  </button>
</div>
`;
